### PR TITLE
Add some criterion benchmarks (stress and binary trees)

### PR DESCRIPTION
**Pull request by Others**
Pretty limited in scope, but a start

### CODE REVIEW

File: benches/shredder_benchmark.rs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+use criterion::{black_box, Criterion, criterion_main};
+use criterion::criterion_group;
+use rand::rngs::StdRng;
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+
+use shredder::{collect, number_of_tracked_allocations, Gc, Scan, run_with_gc_cleanup};
+
+// BENCHMARK 1: My janky stress test
```
Comment (pcodes) [inline benches/shredder_benchmark.rs:11]: Can you add a little description of what it's doing?

File: benches/shredder_benchmark.rs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+struct DirectedGraphNode {
+    label: String,
+    edges: Vec<Gc<RefCell<DirectedGraphNode>>>,
+}
+
+const NODE_COUNT: usize = 1 << 15;
+const EDGE_COUNT: usize = 1 << 15;
+const SHRINK_DIV: usize = 1 << 10;
+
+fn stress_test() -> Vec<usize> {
```
Comment (pcodes) [inline benches/shredder_benchmark.rs:22]: Am I correct in my reading that this is just creating a bunch of nodes, then edges between those nodes, and then removing them?

File: benches/shredder_benchmark.rs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+pub fn benchmark_stress_test(c: &mut Criterion) {
+    c.bench_function("stress_test", |b| b.iter(|| black_box(stress_test())));
+}
+
+
+// BENCHMARK 2: It's binary-trees from the benchmarks game!
+
+fn count_binary_trees(max_size: usize) -> Vec<usize> {
```
Comment (pcodes) [inline benches/shredder_benchmark.rs:61]: Do you ever actually remove anything with this test? Or are you looking to get signal about the overall cleanup?

**Review by pcodes — COMMENTED** [review]
Looks good generally, am I correct in how I've read what the tests are evaluating?

File: benches/shredder_benchmark.rs (hunk 3 above)
Comment (Others) [reply benches/shredder_benchmark.rs:61]: It's supposed to measure mostly allocation timings, automatically triggered collections, and then also the full sweep done by `run_with_gc_cleanup`

File: benches/shredder_benchmark.rs (hunk 2 above)
Comment (Others) [reply benches/shredder_benchmark.rs:22]: Yes

**Pull request merged**